### PR TITLE
Support click track empty body 

### PR DESCRIPTION
**Pull request by namse**
There are 4 commits.
- Support id in event (80f7cc6e7f20fd47e83373619c9f2c5e6a79593d)
  - it's to find out if an empty track body is clicked.
- Mock SequncePlayer (47c8ee5c44d38adc7af803797444328218a178b0)
  - To make a mocked editor in test code.
- Handle shift+click with time (d753e17802c01f74d919a4bbf2f611af0607e2d2)
  - Previously we calculated shift+click using clip ids but it wasn't good idea because there are swallowed clips (example : 1.start--2.start---2.end--1.start).
  - I put test codes together.
- Support click track empty body (14a07a1906102978c79c03e62e7854d1db8d2072)
  - Design doc for this commit : https://docs.google.com/document/d/1TwxQNqeW7EIuoxfoUpfA8fPmDXaXAvONKKIHBJQrX6A/edit#

### CODE REVIEW

**Comment by namse** [comment]
**Rebase merge please**

File: luda-editor/luda-editor-client/src/app/editor/clip_select.rs [hunk 1]
```diff
@@ -61,10 +64,15 @@ impl Editor {
 
         selected_clip_track.get_id() == clip_track.get_id()
     }
-    pub(super) fn select_all_between_clips<T: AsRef<str>>(&mut self, clip_ids: &[T]) {
+    pub(super) fn select_all_to_time(&mut self, time: &Time) {
```
Comment (namse) [inline luda-editor/luda-editor-client/src/app/editor/clip_select.rs:67]: We only think about adding clips. Later, I gonna put the selecting functionality like a text cursor or windows explorer, which support to remove clips from selected clips.

File: luda-editor/luda-editor-client/src/app/editor/sequence_player/mod.rs [hunk 2]
```diff
@@ -57,23 +71,52 @@ impl SequencePlayer {
         this.call_loading_timeout();
         this
     }
-    pub fn play(&mut self) {
+    fn start_play(&mut self) {
```
Comment (namse) [inline luda-editor/luda-editor-client/src/app/editor/sequence_player/mod.rs:74]: Below changes are just order changed. don't care about below.

File: luda-editor/luda-editor-client/src/app/editor/timeline/timeline_body/mod.rs [hunk 3]
```diff
@@ -4,18 +4,54 @@ pub mod track_body;
...
     pub context: &'a TimelineRenderContext<'a>,
 }
+
+struct TimelineBodyLeftClickEvent {
+    pub mouse_event_id: String,
+    pub mouse_position_in_time: Time,
+}
+
 impl TimelineBody {
+    pub(crate) fn new() -> TimelineBody {
```
Comment (namse) [inline luda-editor/luda-editor-client/src/app/editor/timeline/timeline_body/mod.rs:27]: From now on, TimelineBody take care about selected clip binding click event's id to check whether user clicked empty space of timeline body or not.

**Review by bigfoodK — APPROVED** [review]

**Pull request merged**